kratos/tests/test_matrix_interface.py

```diff
@@ -120,5 +120,32 @@ def test_matrix_transpose(self):
             for j in range(A.Size2()):
                 self.assertEqual(A[i,j], B[j,i])
 
+    def test_list_of_list_construction(self):
+        A = KM.Matrix([[2, 2, 2, 2]])                           #List 1
+        self.assertEqual(4, len(A))
+        self.assertEqual(1, A.Size1())
+        self.assertEqual(4, A.Size2())
+        expected = KM.Matrix(1,4)
+        expected.fill(2)
+
+        for i in range(expected.Size1()):
+            for j in range(expected.Size2()):
+                self.assertEqual(A[i,j], 2)
+
+        l2 = [[0, 0.1, 0.2], [1, 1.1, 1.2], [2, 2.1, 2.2]]       #List 2
+        B = KM.Matrix(l2)
+        self.assertEqual(9, len(B))
+        self.assertEqual(3, B.Size1())
+        self.assertEqual(3, B.Size2())
+        expected = KM.Matrix(3,3)
+        for i in range(expected.Size1()):
+            for j in range(expected.Size2()):
+                expected[i,j] = i + (j*0.1)
+
+        for i in range(expected.Size1()):
+            for j in range(expected.Size2()):
+                self.assertEqual(B[i,j], expected[i,j])

```

Comment:
this should be a separate test

Reply:
Why? everything will come under one test for construction of Kratos.Matrix using listoflist?

Reply:
this is such that we can easier detect what exactly fails

Imagine you do five check in one test, then you need to look in detail what actually fails

Ideally tests should be small and only test one thing. Ofc this is not always possible (e.g. for a complete fluid analysis) but in this case it is

This can be considered as a unit test, the complete fluid case would be an integration test (I am not 100% sure abt the naming, google could help to confirm :) )

Reply:
Got it . Thanks :)